providers/mock/mock.go

```diff
@@ -71,6 +71,32 @@ func NewMockProvider(providerConfig, nodeName, operatingSystem string, internalI
 	return &provider, nil
 }
 
+// NewMockProviderMockConfig creates a new MockProvider with the given Mock Config
```

Comment:
Added this just to simplify making a new provider for testing without needing to make files and deal with file io)